src/app/delegates/page.jsx

```diff
@@ -39,7 +40,9 @@ export default async function Page({ searchParams }) {
     <section>
       <Hero />
       <Suspense fallback={<DelegateCardLoadingState />}>
-        <DelegateCardWrapper searchParams={searchParams} />
+        <LayoutProvider>
```

Comment:
We need this LayoutContext to share the state for the picked layout between two sibling components. If we were to do it in a more straightforward way where we just store this state in the parent and pass to its children, we would need to refactor the parent (`DelegateCardWrapper.tsx`) to make it a client component which means moving all async functions one level up. This involves a lot of movement of async functions around. 

I thought this React context approach is simpler in this case. It also benefits us because this layout preference (grid vs table) may be used in other components in the future as well as I think this won't be limited to the Delegates view only. In that case we can just reuse this context without trying to pass the state from the parent.

If there is a strong preference to avoid this React context, I'm happy to turn DelegateCardWrapper into a client component. But like I mentioned, there will be a considerable refactor

Reply:
I don't mind the use of context here -- I agree it would be cumbersome to have to move all of the async calls out.